### ISSUE TITLE
iCloud3 simply stops updating after a while

### ISSUE THREAD

**Issue by lenwar** [gcobb321/icloud3#332]
I'm using 'current versions':
iCloud3 version 3.0.2a
HaOS, running HA 2024.4.3

I use iCloud3 to follow my iDevices on the map for automation purposes. After a while (a number of hours), nothing seems to update anymore. When I restart HA, it works again. "Reloading" the iCloud3 integration doesn't provide any help.

I don't see any concrete error messages. It just stops updating a number of hours (sometimes a day, sometimes 3-4 hours).
How can I troubleshoot this?

```
04-17 12:17:35 ICLOUD ⡇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
                      ⡇   ICLOUD3_DEVICE1, ICLOUD, CURRZONE-NOT_HOME, NEXT UPDATE TIME REACHED 
                      ⡇⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂

04-17 12:17:35 ICLOUD ⡇ icloud3_DEVICE1 > Zone > Away(2) > LOCATION2-185.4m, Thuis(5)-208.4m, LOCATION1-817.7m, LOCATION5-1.27km, LOCATION4-1.46km, LOCATION3-5.70km, DEVICE4 Werk-8.21km, LOCATION6-11.9km, LOCATION7-12.5km, LOCATION8-17.3km, DEVICE2 werk-60.2km, LOCATION9-133km, GPS-(51.85668, 4.31512/±5m)
04-17 12:17:37 ICLOUD ⡇ Waze Results > GPS-(51.85668, 4.31512) --> (52.08582, 5.10909), Time-54.43min, Distance-75.13km
04-17 12:17:37 ICLOUD ⡇ icloud3_DEVICE1 > Waze Route Info > TravTime-54 mins, Dist-75.1km, Moved-55.7m
04-17 12:17:37 ICLOUD ⡇ icloud3_DEVICE1 > Results: From-DEVICE2 werk > Arrive-13:12, NextUpdate-12:44:45, Moved-55.7m (Towards), IntoStatZone-12:23:25, ×Went3km
04-17 12:17:37 ICLOUD ⡇ icloud3_DEVICE1 RESULTS: From-DEVICE2 werk > MobAppZone-not_home, iC3Zone-not_home, Interval-27 mins, TravTime-54 mins, Dist-75.1km, NextUpdt-12:44:45, MaxDist-75.3km, Dir-Towards, Moved-158.2m, Battery-98%, LastDataUpdate-12:17:15, GPSAccuracy-5m, LocAge-36 secs ago, OldThreshold-1 min, LastEvLogMsg-12:17:37, Method-3.WazeTime
04-17 12:17:38 ICLOUD ⡇ Waze Results > GPS-(51.85668, 4.31512) --> (51.89780, 4.55716), Time-25.35min, Distance-27.01km
04-17 12:17:38 ICLOUD ⡇ icloud3_DEVICE1 > Waze Route Info > TravTime-25 mins, Dist-27.0km, Moved-55.7m (Saved)
04-17 12:17:38 ICLOUD ⡇ icloud3_DEVICE1 > Results: From-LOCATION8 > Arrive-12:42, NextUpdate-12:30:15, Moved-55.7m (Towards), IntoStatZone-12:23:25, ×Went3km
04-17 12:17:38 ICLOUD ⡇ icloud3_DEVICE1 RESULTS: From-LOCATION8 > MobAppZone-not_home, iC3Zone-not_home, Interval-13 mins, TravTime-25 mins, Dist-27.0km, NextUpdt-12:30:15, MaxDist-27.2km, Dir-Towards, Moved-158.2m, Battery-98%, LastDataUpdate-12:17:15, GPSAccuracy-5m, LocAge-37 secs ago, OldThreshold-1 min, LastEvLogMsg-12:17:38, Method-3.WazeTime
04-17 12:17:38 ICLOUD ⡇ icloud3_DEVICE1 > Results: From-Thuis > Arrive-12:17, NextUpdate-12:18:35, IntoStatZone-12:23:25, Battery-98%, ×Went3km, WazeMsg-×Under1km
04-17 12:17:38 ICLOUD ⡇ icloud3_DEVICE1 RESULTS: From-Thuis > MobAppZone-not_home, iC3Zone-not_home, Interval-1 min, TravTime-0 min, Dist-208.4m, NextUpdt-12:18:35, MaxDist-0.0km, Dir-Towards, Moved-158.2m, Battery-98%, LastDataUpdate-12:17:15, GPSAccuracy-5m, LocAge-37 secs ago, OldThreshold-1 min, LastEvLogMsg-12:17:38, Method-3.<3km
04-17 12:17:38 ICLOUD ⡇⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂
                      ⡇   ICLOUD3_DEVICE1, FAMSHR, CURRZONE-NOT_HOME,  
                      ⡇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
04-17 12:17:45 LOCATE   icloud3_apple_watch_van_DEVICE2 > Located > FamShr-12:17:03 (42 secs ago)
04-17 12:17:45 ICLOUD   icloud3_apple_watch_van_DEVICE2 > Trigger > Next Update Time Reached
04-17 12:17:45 ICLOUD ⡇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
                      ⡇   ICLOUD3_APPLE_WATCH_VAN_DEVICE2, ICLOUD, CURRZONE-NOT_HOME, NEXT UPDATE TIME REACHED 
                      ⡇⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂

04-17 12:17:45 ICLOUD ⡇ icloud3_apple_watch_van_DEVICE2 > Zone > Away(2) > LOCATION2-185.4m, Thuis(5)-208.4m, LOCATION1-817.7m, LOCATION5-1.27km, LOCATION4-1.46km, LOCATION3-5.70km, DEVICE4 Werk-8.21km, LOCATION6-11.9km, LOCATION7-12.5km, LOCATION8-17.3km, DEVICE2 werk-60.2km, LOCATION9-133km, GPS-(51.85668, 4.31512/±5m)
04-17 12:17:46 ICLOUD ⡇ Waze Results > GPS-(51.85668, 4.31512) --> (52.08582, 5.10909), Time-54.48min, Distance-75.13km
04-17 12:17:46 ICLOUD ⡇ icloud3_apple_watch_van_DEVICE2 > Waze Route Info > TravTime-54 mins, Dist-75.1km, Moved-24.8m
04-17 12:17:46 ICLOUD ⡇ icloud3_apple_watch_van_DEVICE2 > Results: From-DEVICE2 werk > Arrive-13:12, NextUpdate-12:44:55, Moved-24.8m (Towards), IntoStatZone-12:24:45, ×Went3km
04-17 12:17:46 ICLOUD ⡇ icloud3_apple_watch_van_DEVICE2 RESULTS: From-DEVICE2 werk > MobAppZone-not_set, iC3Zone-not_home, Interval-27 mins, TravTime-54 mins, Dist-75.1km, NextUpdt-12:44:55, MaxDist-75.3km, Dir-Towards, Moved-24.8m, Battery-100%, LastDataUpdate-12:17:15, GPSAccuracy-5m, LocAge-43 secs ago, OldThreshold-1 min, LastEvLogMsg-12:17:46, Method-3.WazeTime
04-17 12:17:46 ICLOUD ⡇ icloud3_apple_watch_van_DEVICE2 > Waze Route Info > Using Route History Database (#16)
04-17 12:17:46 ICLOUD ⡇ icloud3_apple_watch_van_DEVICE2 > Results: From-LOCATION8 > Arrive-12:43, NextUpdate-12:30:25, Moved-24.8m (Towards), IntoStatZone-12:24:45, ×Went3km
04-17 12:17:46 ICLOUD ⡇ icloud3_apple_watch_van_DEVICE2 RESULTS: From-LOCATION8 > MobAppZone-not_set, iC3Zone-not_home, Interval-13 mins, TravTime-25 mins, Dist-27.0km, NextUpdt-12:30:25, MaxDist-27.2km, Dir-Towards, Moved-24.8m, Battery-100%, LastDataUpdate-12:17:15, GPSAccuracy-5m, LocAge-43 secs ago, OldThreshold-1 min, LastEvLogMsg-12:17:46, Method-3.WazeTime
04-17 12:17:46 ICLOUD ⡇ icloud3_apple_watch_van_DEVICE2 > Results: From-Thuis > Arrive-12:17, NextUpdate-12:18:45, IntoStatZone-12:24:45, Battery-100%, ×Went3km, WazeMsg-×Under1km
04-17 12:17:46 ICLOUD ⡇ icloud3_apple_watch_van_DEVICE2 RESULTS: From-Thuis > MobAppZone-not_set, iC3Zone-not_home, Interval-1 min, TravTime-0 min, Dist-208.4m, NextUpdt-12:18:45, MaxDist-0.0km, Dir-Towards, Moved-24.8m, Battery-100%, LastDataUpdate-12:17:15, GPSAccuracy-5m, LocAge-43 secs ago, OldThreshold-1 min, LastEvLogMsg-12:17:46, Method-3.<3km
04-17 12:17:46 ICLOUD ⡇⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂
                      ⡇   ICLOUD3_APPLE_WATCH_VAN_DEVICE2, FAMSHR, CURRZONE-NOT_HOME,  
                      ⡇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

04-17 12:22:21 ICLOUD   Service Action > Action-Send, Options-locate request to mobile app
04-17 12:22:32 ICLOUD   Service Action > Action-Locate, Options-device(s) using icloud famshr
04-17 12:22:32 ICLOUD   icloud3_DEVICE1 > Location will be updated at 12:22:35
04-17 12:22:32 ICLOUD   icloud3_apple_watch_van_DEVICE2 > Location will be updated at 12:22:35
04-17 12:22:32 ICLOUD   icloud3_iphone_van_DEVICE4 > Location will be updated at 12:22:35
04-17 12:22:32 ICLOUD   icloud3_ipad_van_DEVICE4 > Location will be updated at 12:22:35
04-17 12:22:32 ICLOUD   icloud3_iphone_van_DEVICE3 > Location will be updated at 12:22:35
04-17 12:22:32 ICLOUD   icloud3_iphone_van_DEVICE5 > Location will be updated at 12:22:35
04-17 12:22:32 ICLOUD   icloud3_iphone_van_DEVICE2_werk > Location will be updated at 12:22:35
04-17 12:23:05 ICLOUD   Service Action > Action-Send, Options-locate request to mobile app
04-17 12:23:15 ICLOUD   Service Action > Action-Locate, Options-device(s) using icloud famshr
04-17 12:23:15 ICLOUD   icloud3_DEVICE1 > Location will be updated at 12:23:15
04-17 12:23:15 ICLOUD   icloud3_apple_watch_van_DEVICE2 > Location will be updated at 12:23:15
04-17 12:23:15 ICLOUD   icloud3_iphone_van_DEVICE4 > Location will be updated at 12:23:15
04-17 12:23:15 ICLOUD   icloud3_ipad_van_DEVICE4 > Location will be updated at 12:23:20
04-17 12:23:15 ICLOUD   icloud3_iphone_van_DEVICE3 > Location will be updated at 12:23:20
04-17 12:23:15 ICLOUD   icloud3_iphone_van_DEVICE5 > Location will be updated at 12:23:20
04-17 12:23:15 ICLOUD   icloud3_iphone_van_DEVICE2_werk > Location will be updated at 12:23:20
04-17 12:23:38 ICLOUD   Service Action > Action-Send, Options-locate request to mobile app
04-17 12:23:48 ICLOUD   Service Action > Action-Locate, Options-device(s) using icloud famshr
```

**Comment by gcobb321** [comment]
Turn on log_level logging.  _Event Log > Actions > Start Debug Logging_ will turn it on for a day (it will reert back to Info at midnight). Doing it from _Configure Settings > Format Parameters_ screen will turn it on until you turn it off.

That might show more info. 

Are there any errors in the _homeassistant.log_ file?
Does anything show up in the Event Log during startup? 
Maybe change the watch and ipad to Tracking Mode=Monitor for a while and see if it is device related.

**Comment by lenwar** [comment]
Hi,

I set it to debug, but I couldn't find anything exciting. Also there are no errors related to icloud3 in the home-assistant logs.

I tried to change all devices without a home-assistant app running (my watch and two old phones which are no long supported by home-assistant due to the older version of iOS).

And that seemed to help. It's running okay for a few days now.

I'll keep trying to isolate that behaviour further and keep you posted. Maybe there is summer kind of bug for when a phone-device either doesn't have the app, and/or is running an old version of iOS?

Anyway. I'll check in again after a few days.
So I'll start changing the watch and tablet back to 'track' and see if that makes any change z
